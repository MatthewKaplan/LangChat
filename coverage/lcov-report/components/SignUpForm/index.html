<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/SignUpForm</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/SignUpForm
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>23/60</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.62% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.6% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/57</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="SignUpForm.js"><a href="SignUpForm.js.html">SignUpForm.js</a></td>
	<td data-value="38.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 38%;"></div><div class="cover-empty" style="width:62%;"></div></div></td>
	<td data-value="38.33" class="pct low">38.33%</td>
	<td data-value="60" class="abs low">23/60</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="18" class="abs low">6/18</td>
	<td data-value="34.62" class="pct low">34.62%</td>
	<td data-value="26" class="abs low">9/26</td>
	<td data-value="38.6" class="pct low">38.6%</td>
	<td data-value="57" class="abs low">22/57</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jul 22 2019 01:53:48 GMT-0600 (Mountain Daylight Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
