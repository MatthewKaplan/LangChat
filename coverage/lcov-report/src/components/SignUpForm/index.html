<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/SignUpForm</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/SignUpForm
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
<<<<<<< HEAD
        <span class="strong">90.16% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>55/61</span>
=======
<<<<<<< HEAD
        <span class="strong">88.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>53/60</span>
=======
        <span class="strong">88.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>52/59</span>
>>>>>>> master
>>>>>>> 6bba9738226438400dee617c6f3d992ce1d88c0d
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.44% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>17/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.86% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>26/28</span>
      </div>
      <div class='fl pad1y space-right2'>
<<<<<<< HEAD
        <span class="strong">91.53% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>54/59</span>
=======
<<<<<<< HEAD
        <span class="strong">89.66% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>52/58</span>
=======
        <span class="strong">89.47% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>51/57</span>
>>>>>>> master
>>>>>>> 6bba9738226438400dee617c6f3d992ce1d88c0d
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="SignUpForm.js"><a href="SignUpForm.js.html">SignUpForm.js</a></td>
<<<<<<< HEAD
	<td data-value="90.16" class="pic high"><div class="chart"><div class="cover-fill" style="width: 90%;"></div><div class="cover-empty" style="width:10%;"></div></div></td>
	<td data-value="90.16" class="pct high">90.16%</td>
	<td data-value="61" class="abs high">55/61</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="28" class="abs high">26/28</td>
	<td data-value="91.53" class="pct high">91.53%</td>
	<td data-value="59" class="abs high">54/59</td>
=======
<<<<<<< HEAD
	<td data-value="88.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.33" class="pct high">88.33%</td>
	<td data-value="60" class="abs high">53/60</td>
=======
	<td data-value="88.14" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.14" class="pct high">88.14%</td>
	<td data-value="59" class="abs high">52/59</td>
>>>>>>> master
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
<<<<<<< HEAD
	<td data-value="89.66" class="pct high">89.66%</td>
	<td data-value="58" class="abs high">52/58</td>
=======
	<td data-value="89.47" class="pct high">89.47%</td>
	<td data-value="57" class="abs high">51/57</td>
>>>>>>> master
	</tr>

<tr>
	<td class="file empty" data-value="styles.js"><a href="styles.js.html">styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
>>>>>>> 6bba9738226438400dee617c6f3d992ce1d88c0d
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
<<<<<<< HEAD
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 25 2019 01:42:18 GMT-0600 (Mountain Daylight Time)
=======
<<<<<<< HEAD
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jul 24 2019 23:12:14 GMT-0600 (Mountain Daylight Time)
=======
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jul 24 2019 22:34:25 GMT-0600 (Mountain Daylight Time)
>>>>>>> master
>>>>>>> 6bba9738226438400dee617c6f3d992ce1d88c0d
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
