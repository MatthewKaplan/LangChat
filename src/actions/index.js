export const changeEmail = text => ({
	type: 'CHANGE_EMAIL',
	text
});

export const changePassword = text => ({
	type: 'CHANGE_PASSWORD',
	text
});

export const addMessage = message => ({
	type: 'ADD_MESSAGE',
	payload: { message }
});
