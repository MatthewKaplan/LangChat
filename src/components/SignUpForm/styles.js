import { StyleSheet } from 'react-native';

export default StyleSheet.create({
	inputContainerStyle: {
		backgroundColor: '#fff',
		alignSelf: 'center',
		flexDirection: 'column',
		width: '90%',
		margin: 10,
		borderRadius: 5
	},
	container: {
		backgroundColor: '#323232',
		flex: 1,
		width: '100%',
		shadowColor: '#000',
		shadowOffset: { width: 0, height: 2 },
		shadowOpacity: 0.1,
		shadowRadius: 2,
		elevation: 1,
		paddingTop: 50,
		position: 'relative'
	},
	textHeaderStyle: {
		fontSize: 50,
		fontWeight: '600',
		color: 'white',
		margin: 40,
		textAlign: 'center',
		textTransform: 'uppercase',
		fontWeight: 'bold'
	},
	buttonContainerStyle: {
		// marginTop: 80
	}
});
