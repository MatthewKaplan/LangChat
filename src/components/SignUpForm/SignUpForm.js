import React, { Component } from 'react';
import { View, Text, Picker } from 'react-native';
import { Actions } from 'react-native-router-flux';
import Button from '../common/Button';
import Input from '../common/Input';
import { MaterialCommunityIcons, AntDesign, EvilIcons, Feather } from '@expo/vector-icons';
import Data from '../../Helper/data';
import { connect } from 'react-redux';
import {
	createEmail,
	createPassword,
	createFirstName,
	createLastName,
	createUserName,
	createCountry,
	login
} from '../../actions';
import styles from './styles';

export class SignUpForm extends Component {
	state = {
		userInfo: true,
		userCountry: false,
		userCredentials: false,
		confirmation: '',
		error: ''
	};

	buttonToRender = () => {
		const { firstName, lastName, userName, email, password } = this.props;
		if (this.state.userInfo) {
			return (
				<Button
					disabled={!firstName || !lastName || !userName}
					onPress={() => this.setState({ userInfo: false, userCountry: true })}
					data-test="user-name-btn">
					Next
				</Button>
			);
		}
		if (this.state.userCountry) {
			return (
				<Button onPress={() => this.setState({ userCountry: false, userCredentials: true })} data-test="next-btn">
					Next
				</Button>
			);
		}
		return (
			<Button
				disabled={(!email || !password) && this.validatePassword()}
				onPress={this.register}
				data-test="email-password-submit-btn">
				Sign Up!
			</Button>
		);
	};

	renderPicker = () => {
		const countries = Data.countries;
		const { createCountry, country } = this.props;
		return (
			<Picker
				style={{
					width: '100%',
					height: 244
				}}
				itemStyle={{ fontSize: 25 }}
				selectedValue={country}
				onValueChange={itemValue => createCountry(itemValue)}
				data-test="country-picker">
				{countries.map(country => {
					return <Picker.Item key={country} label={country} value={country} />;
				})}
			</Picker>
		);
	};

	register = async () => {
		const { email, password, firstName, lastName, userName, country } = this.props;
		const newUser = {
			displayName: userName,
			email,
			firstName,
			lastName,
			active: true,
			countryOfOrigin: country,
			password,
			passwordConfirmation: password
		};

		try {
			const res = await fetch('https://langchat-crosspollination.herokuapp.com/api/v1/users/', {
				method: 'POST',
				headers: { 'Content-Type': 'application/json' },
				body: JSON.stringify(newUser)
			});
			const user = await res.json();
			this.login(user);
			// Actions.loginForm();
		} catch (error) {
			this.setState({ error: 'Unable to register new user.' });
		}
	};

	login = async user => {
		const { username } = user;
		try {
			const res = await fetch('https://langchat-crosspollination.herokuapp.com/api/v1/log_in/?format=json', {
				method: 'POST',
				headers: { 'Content-Type': 'application/json', Accept: 'application/json' },
				body: JSON.stringify({ username, password: this.props.password })
			});
			const user = await res.json();
			const cookieData = res.headers.get('set-cookie'); //use postive lookbehind to extract csfrtoken= and positive lookahead to extract ;
			const match = cookieData.match(/(csrftoken=)\w+;/);
			const csrftoken = match[0].split('=')[1].slice(0, -1);
			this.props.login(user, csrftoken); // redux
			Actions.tutorial();
		} catch (error) {
			console.log(error.message);
		}
	};

	validatePassword = () => {
		return this.state.confirmation !== this.props.password;
	};

	render () {
		const { container, inputContainerStyle, textHeaderStyle } = styles;
		const {
			email,
			password,
			firstName,
			lastName,
			createFirstName,
			createLastName,
			createEmail,
			createPassword,
			userName,
			createUserName
		} = this.props;
		return (
			<View style={container}>
				<EvilIcons
					name="close"
					size={40}
					onPress={() => Actions.splashPage()}
					style={{ width: '13%', alignSelf: 'flex-end', color: 'white' }}
					data-test="close-btn"
				/>
				<Text style={textHeaderStyle}>Sign Up</Text>
				{this.state.userInfo && (
					<React.Fragment>
						<View style={inputContainerStyle}>
							<Input
								label={<Feather name="plus" size={30} color="#999" />}
								placeholder="First Name"
								value={firstName}
								onChangeText={firstName => createFirstName(firstName)}
								data-test="first-name-input"
							/>
						</View>
						<View style={inputContainerStyle}>
							<Input
								label={<Feather name="plus" size={30} color="#999" />}
								placeholder="Last Name"
								value={lastName}
								onChangeText={lastName => createLastName(lastName)}
								data-test="last-name-input"
							/>
						</View>
						<View style={inputContainerStyle}>
							<Input
								label={<AntDesign name="user" size={30} color="#999" />}
								placeholder="Username"
								value={userName}
								onChangeText={userName => createUserName(userName)}
								data-test="user-input"
							/>
						</View>
					</React.Fragment>
				)}
				{this.state.userCountry && <View style={inputContainerStyle}>{this.renderPicker()}</View>}
				{this.state.userCredentials && (
					<React.Fragment>
						<View style={inputContainerStyle}>
							<Input
								label={<MaterialCommunityIcons name="email-outline" size={30} color="#999" />}
								placeholder="Email"
								value={email}
								onChangeText={email => createEmail(email)}
								data-test="email-input"
							/>
						</View>
						<View style={inputContainerStyle}>
							<Input
								label={<AntDesign name="lock" size={30} color="#999" />}
								placeholder="Password"
								secureTextEntry
								value={password}
								onChangeText={password => createPassword(password)}
								data-test="password-input"
							/>
						</View>
						<View style={inputContainerStyle}>
							<Input
								label={<AntDesign name="lock" size={30} color="#999" />}
								placeholder="Confirm Password"
								value={this.state.confirmation}
								onChangeText={text => this.setState({ confirmation: text })}
								secureTextEntry
								data-test="confirm-password-input"
							/>
						</View>
						{this.validatePassword() && <Text>Passwords do not match</Text>}
					</React.Fragment>
				)}
				{this.buttonToRender()}
			</View>
		);
	}
}

export const mapStateToProps = state => ({
	email: state.register.email,
	password: state.register.password,
	firstName: state.register.firstName,
	lastName: state.register.lastName,
	userName: state.register.userName,
	country: state.register.country
});

export const mapDispatchToProps = dispatch => ({
	createFirstName: firstName => dispatch(createFirstName(firstName)),
	createLastName: lastName => dispatch(createLastName(lastName)),
	createEmail: email => dispatch(createEmail(email)),
	createPassword: password => dispatch(createPassword(password)),
	createUserName: userName => dispatch(createUserName(userName)),
	createCountry: country => dispatch(createCountry(country)),
	login: (user, cookie) => dispatch(login(user, cookie))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUpForm);
