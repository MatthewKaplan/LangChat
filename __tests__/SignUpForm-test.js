// __tests__/SignUpForm-test.js
import React from 'react';
import { SignUpForm, mapStateToProps, mapDispatchToProps } from '../src/components/SignUpForm/SignUpForm';
import * as actions from '../src/actions/index';

import renderer from 'react-test-renderer';

describe('SignUpForm Component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<SignUpForm />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should have default state', () => {
		const wrapper = renderer.create(<SignUpForm />).getInstance();

		expect(wrapper.state).toEqual({
			userInfo: true,
			userCountry: false,
			userCredentials: false,
			confirmation: '',
			error: ''
		});
	});
});

// describe('mapStateToProps', () => {
// 	it('should return an object', () => {
// 		const mockData = {
// 			email: '',
// 			password: '',
// 			firstName: '',
// 			lastName: '',
// 			userName: '',
// 			country: ''
// 		};
// 		const expected = {
// 			email: '',
// 			password: '',
// 			firstName: '',
// 			lastName: '',
// 			userName: '',
// 			country: ''
// 		};

// 		const mockprops = mapStateToProps(mockData);
// 		expect(mockprops).toEqual(expected);
// 	});
// });

describe('mapDispatchToProps', () => {
	it('should call dispatch for createFirstName', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.createFirstName('mockFirstName');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.createFirstName('mockFirstName');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for createLastName', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.createLastName('mockLastName');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.createLastName('mockLastName');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for createEmail', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.createEmail('mockEmail');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.createEmail('mockEmail');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for createPassword', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.createPassword('mockPassword');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.createPassword('mockPassword');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for createUserName', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.createUserName('mockUserName');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.createUserName('mockUserName');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for createCountry', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.createCountry('mockCountry');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.createCountry('mockCountry');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});
});
