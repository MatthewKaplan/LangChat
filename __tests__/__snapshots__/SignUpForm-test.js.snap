// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm Component should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#323232",
      "elevation": 1,
      "flex": 1,
      "paddingTop": 50,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 50,
        "fontWeight": "bold",
        "margin": 40,
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Sign Up
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flexDirection": "column",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="First Name"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "height": "100%",
            "lineHeight": 23,
            "margin": "auto",
            "paddingLeft": 5,
            "paddingRight": 5,
            "width": "95%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flexDirection": "column",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Last Name"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "height": "100%",
            "lineHeight": 23,
            "margin": "auto",
            "paddingLeft": 5,
            "paddingRight": 5,
            "width": "95%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "flexDirection": "column",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Username"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "height": "100%",
            "lineHeight": 23,
            "margin": "auto",
            "paddingLeft": 5,
            "paddingRight": 5,
            "width": "95%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#007aff",
        "borderRadius": 5,
        "margin": 10,
        "opacity": 1,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 19,
          "fontWeight": "600",
          "paddingBottom": 12,
          "paddingTop": 12,
        }
      }
    >
      Next
    </Text>
  </View>
</View>
`;
